e2e/utils/upload-database.ts

```diff
@@ -0,0 +1,37 @@
+import { expect } from '@playwright/test'
+import path from "path";
+
+const dbNamePrefix = 'test/'
+const dbName = 'empty-db'
+const dbExtension = '.sql.gz'
+
+export const uploadTestDatabase = async (page) => {
+    await page.getByRole('link', { name: 'Databases' }).click()
+    await expect(page.getByRole('button', { name: 'Upload database' })).toBeVisible()
+    // TODO find a better way to make sure the page is loaded and if there was an test db uploaded it's shown.
+    // The problem is that if there are no databases uploaded, only the "Upload database" button is displayed
+    // and it loads before the databases list table, so it's unreliable to just wait for the button to be visible.
+    // page.waitForLoadState() with any of the state options doesn't work.
```

Comment:
I hope that what I am about to write is of any use, because it's going to be a bit of a StackOverflow style answer, not at all providing a solution but rather redefining the problem....

1. Ideally we start with a clean backend, so we always _have to_ create this test database at the start of the run. And since the backend is pristine, we _know_ that it hasn't been created yet. If this was the case, we didn't have to have any conditional logic in place.
2. Uploading the test database is not currently part of the test suite. And even it if was to become part of it, we shouldn't rely on the side effects of one test in other tests. As such I'd say this is more part of the test suite's setup process. So we need to probably do this in one of the ways outlined in the [Global setup and teardown docs](https://playwright.dev/docs/test-global-setup-teardown).
3. Since this is not part of the test suite itself, I guess it could be cleaner to upload the file using a HTTP request. However, doing it via the UI seems to be very acceptable too going by the docs.

Reply:
So in summary I suppose the way to go here would be to upload the database as part of the global [Setup](https://playwright.dev/docs/test-global-setup-teardown#setup) and then delete it as part of the global [Teardown](https://playwright.dev/docs/test-global-setup-teardown#setup), right? 🤔 

@tonsV2 Any thoughts on this?

Reply:
What you're suggesting _could_ be the right approach. But it depends: 
1. If we need this DB for one test only, then I would actually recommend using the `beforeAll` and `afterAll` hooks of the test that requires the DB (see [docs](https://playwright.dev/docs/api/class-test#test-before-all-1)).
2. If we (expect to) need this DB for more tests, then using the global setup and teardown seems more appropriate.

Reply:
Alright, will stick with `beforeAll` and `afterAll` for now. 👍